src/main/java/org/jenkinsci/plugins/oic/OicSecurityRealm.java

```diff
@@ -887,6 +896,21 @@ private List<String> ensureString(Object field) {
                 }
             }
             return result;
+        } else if (field instanceof ArrayList) {
+            List<String> result = new ArrayList<>();
+            List<Object> groups = (List<Object>) field;
+            for (Object group : groups) {
+              if (group instanceof String) {
+                result.add(group.toString());
+              } else if (group instanceof ArrayMap) {
+                // if its a Map, we use the nestedGroupFieldName to grab the groups
+                Map<String, String> groupMap = (Map<String, String>) group;
+                if (groupMap.keySet().contains(nestedGroupFieldName)) {
```

Comment:
Test nestedGroupFieldName not null (with new logic)

Reply:
updated in latest changes